File: structural/facade/typescript/facade.ts
import { DVDPlayer, SoundSystem, TV } from './subsystems';

export class HomeEntertainmentFacade {
	private tv: TV;
	private dvdPlayer: DVDPlayer;
	private soundSystem: SoundSystem;

	constructor() {
		this.tv = new TV();
		this.dvdPlayer = new DVDPlayer();
		this.soundSystem = new SoundSystem();
	}

	watchMovie(channel: string, volume: number = 20): void {
		this.tv.turnOn();
		this.soundSystem.turnOn();
		this.dvdPlayer.turnOn();

		this.tv.changeChannel(channel);

		this.dvdPlayer.play();

		this.soundSystem.setVolume(volume);
	}

	stopMovie(): void {
		this.dvdPlayer.stop();

		this.tv.turnOff();
		this.soundSystem.turnOff();
		this.dvdPlayer.turnOff();
	}
}

File: structural/facade/python/facade.py
from subsystems import TV, DVDPlayer, SoundSystem


class HomeEntertainmentFacade:
    def __init__(self):
        self.tv = TV()
        self.dvd_player = DVDPlayer()
        self.sound_system = SoundSystem()

    def watch_movie(self, channel, volume=20):
        self.tv.turn_on()
        self.sound_system.turn_on()
        self.dvd_player.turn_on()
        self.tv.change_channel(channel)
        self.dvd_player.play()
        self.sound_system.set_volume(volume)

    def stop_movie(self):
        self.dvd_player.stop()
        self.tv.turn_off()
        self.sound_system.turn_off()
        self.dvd_player.turn_off()
